solr/core/src/java/org/apache/solr/client/solrj/embedded/EmbeddedSolrServer.java

```diff
@@ -271,7 +273,9 @@ public void writeResults(ResultContext ctx, JavaBinCodec codec) throws IOExcepti
       throw new SolrServerException(ex);
     } finally {
       if (req != null) req.close();
-      SolrRequestInfo.clearRequestInfo();
+      if (mustClearSolrRequestInfo) {
```

Comment:
Hmm; might it simply be enough to call clearRequestInfo if req != null, thus no new boolean?

Reply:
actually that's enough